vllm/entrypoints/llm.py

```diff
@@ -262,6 +307,19 @@ def _validate_and_prepare_requests(
         if isinstance(prompts, str):
             # Convert a single prompt to a list.
             prompts = [prompts]
+        elif isinstance(prompts[0], dict):
+            # Apply chat templates for chat inputs.
+            tokenizer = self.get_tokenizer()
+            prompts = tokenizer.apply_chat_template(
+                prompts, tokenize=False, add_generation_template=True)
```

Comment:
Do we always want to enable `add_generation_template`?

Reply:
I added an optional parameter for that.